qiskit_ibm_runtime/_cli.py

```diff
@@ -0,0 +1,264 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+The `save-account` command-line interface.
+
+These classes and functions are not public.
+"""
+
+import argparse
+import sys
+from getpass import getpass
+from typing import List, Literal, Callable, TypeVar
+
+from ibm_cloud_sdk_core.api_exception import ApiException
+
+from .qiskit_runtime_service import QiskitRuntimeService
+from .exceptions import IBMNotAuthorizedError
+from .api.exceptions import RequestsApiError
+from .accounts.management import AccountManager, _DEFAULT_ACCOUNT_CONFIG_JSON_FILE
+from .accounts.exceptions import AccountAlreadyExistsError
+
+Channel = Literal["ibm_quantum", "ibm_cloud"]
+T = TypeVar("T")
+
+
+def entry_point() -> None:
+    """
+    This is the entry point for the `qiskit-ibm-runtime` command. At the
+    moment, we only support one script (save-account), but we want to have a
+    `qiskit-ibm-runtime` command so users can run `pipx run qiskit-ibm-runtime
+    save-account`.
+    """
+    # Use argparse to create the --help feature
+    parser = argparse.ArgumentParser(
+        prog="qiskit-ibm-runtime",
+        description="Commands for the Qiskit IBM Runtime Python package",
+    )
+    parser.add_subparsers(
+        title="Commands",
+        description="This package supports the following commands:",
+        dest="script",
+        required=True,
+    ).add_parser(
+        "save-account",
+        description=(
+            "An interactive command-line interface to save your Qiskit IBM "
+            "Runtime account locally. This script is interactive-only."
+        ),
+        help="Interactive command-line interface to save your account locally.",
+    ).add_argument(
+        "--no-color", action="store_true", help="Hide ANSI escape codes in output"
+    )
+    args = parser.parse_args()
+    use_color = not args.no_color
+    if args.script == "save-account":
+        try:
+            SaveAccountCLI(color=use_color).main()
+        except KeyboardInterrupt:
+            sys.exit()
+
+
+class Formatter:
+    """Format using terminal escape codes"""
+
+    # pylint: disable=missing-function-docstring
+    #
+    def __init__(self, color: bool):
+        self.color = color
+
+    @staticmethod
+    def _skip_if_no_color(
+        method: Callable[["Formatter", str], str]
+    ) -> Callable[["Formatter", str], str]:
```

Comment:
Rather than a decorator, a much simpler approach is to have a staticmethod called `_color()`. Have it take `s` and `color_code`, which is either 1, 32, 31, or 36. Note that everything else is the same in the ansi codes.

Keep it simple.

--

Feel free to use a different name of course. Maybe rename the instance var `color` to `color_enabled`

Reply:
Ah yes that is much simpler, thanks